src/Tests/Microsoft.NET.Publish.Tests/RuntimeIdentifiersTests.cs

```diff
@@ -193,6 +193,76 @@ public void PublishWithRuntimeIdentifier(bool publishNoBuild)
             }
         }
 
+        [WindowsOnlyTheory]
+        [InlineData("net7.0", "win-x64", "win-x86", false, false)]
+        [InlineData("net7.0", "win-x64", "win-x86", true, false)]
+        [InlineData("net7.0", "win-x64", "win-x86", true, true)]
+        public void PublishRuntimeIdentifierSetsRuntimeIdentifierAndDoesOrDoesntOverrideRID(string tfm, string publishRuntimeIdentifier, string runtimeIdentifier, bool runtimeIdentifierIsGlobal, bool publishRuntimeIdentifierIsGlobal)
+        {
+            var testProject = new TestProject()
+            {
+                IsExe = true,
+                TargetFrameworks = tfm
+            };
+            testProject.AdditionalProperties["RuntimeIdentifier"] = runtimeIdentifier;
+            testProject.AdditionalProperties["PublishRuntimeIdentifier"] = publishRuntimeIdentifier;
+            testProject.RecordProperties("RuntimeIdentifier");
+
+            List<string> args = new List<string>
+            {
+                $"/p:_IsPublishing=true", // Normally this would be set by the CLI (OR VS Soon TM), but this calls directly into t:/Publish.
+                runtimeIdentifierIsGlobal ? $"/p:RuntimeIdentifier={runtimeIdentifier}" : "",
+                publishRuntimeIdentifierIsGlobal ? $"/p:PublishRuntimeIdentifier={publishRuntimeIdentifier}" : ""
+            };
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: $"{publishRuntimeIdentifierIsGlobal}-{runtimeIdentifierIsGlobal}");
```

Comment:
We should include all of the theory parameters in the identifier here.  If some values don't change (like the RuntimeIdentifiers), then we may not need to include them as test theory parameters at all, we can just hard-code them in the test.